src/repository.ts

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import * as os from 'os';
+import * as glob from 'glob';
+import { pwd } from 'shelljs';
+import { AnyJson } from '@salesforce/ts-types';
+import { UX } from '@salesforce/command';
+import { exec, ShellString } from 'shelljs';
+import { fs, Logger, SfdxError } from '@salesforce/core';
+import { AsyncOptionalCreatable, Env } from '@salesforce/kit';
+import { get, Nullable } from '@salesforce/ts-types';
+import * as chalk from 'chalk';
+import { api as packAndSignApi, SigningResponse } from './codeSigning/packAndSign';
+import { upload } from './codeSigning/upload';
+import { Package, VersionValidation } from './package';
+
+type LernaJson = {
+  packages?: string[];
+} & AnyJson;
+
+interface PrepareOpts {
+  dryrun?: boolean;
+}
+
+interface PublishOpts {
+  dryrun?: boolean;
+  signatures?: SigningResponse[];
+  tag?: string;
+  access?: 'public' | 'restricted';
+}
+
+interface VersionsByPackage {
+  [key: string]: {
+    currentVersion: string;
+    nextVersion: string;
+  };
+}
+
+export async function isMonoRepo(): Promise<boolean> {
+  return fs.fileExists('lerna.json');
+}
+
+async function readLernaJson(): Promise<LernaJson> {
+  return (await fs.readJson('lerna.json')) as LernaJson;
+}
+
+async function wait(ms: number): Promise<void> {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+}
+
+async function getListOfPackages(): Promise<string[]> {
+  const workingDir = pwd().stdout;
+  const lernaJson = await readLernaJson();
+  const packageGlobs = lernaJson.packages || ['*'];
+  const packages = packageGlobs
+    .map((pGlob) => glob.sync(pGlob))
+    .reduce((x, y) => x.concat(y), [])
+    .map((pkg) => path.join(workingDir, pkg));
+  return packages;
+}
+
+class Signer extends AsyncOptionalCreatable {
+  public static PUBLIC_KEY_URL = 'https://developer.salesforce.com/media/salesforce-cli/sfdx-cli-03032020.crt';
+  public static SIGNATURE_URL = 'https://developer.salesforce.com/media/salesforce-cli/signatures';
+  public keyPath: string;
+  private logger: Logger;
+  private ux: UX;
+
+  public constructor(ux: UX) {
+    super(ux);
+    this.ux = ux;
+  }
+
+  public async prepare(): Promise<void> {
+    const fingerprint = await packAndSignApi.retrieveFingerprint(Signer.PUBLIC_KEY_URL);
+    const env = new Env();
+    env.setString('SFDX_DEVELOPER_TRUSTED_FINGERPRINT', fingerprint);
+
+    const key = Buffer.from(env.getString('SALESFORCE_KEY'), 'base64').toString();
+    this.keyPath = path.join(os.tmpdir(), 'salesforce-cli.key');
```

Comment:
do we even need to store the file now? can't we just always read from the env var?

Reply:
@amphro We don't have to store the file anymore but I was hoping to refactor the signing code as little as possible